### PR TITLE
HyperTableV2::Selection component

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

Add HyperTableV2::Selection component:
- implement it
- add tests
- add demo in `application.hbs`

Related to : https://github.com/upfluence/backlog/issues/2317

### What are the observable changes?
Peek 2023-01-31 10-15

### 🧑‍💻 Developer Heads Up

⚡ Since we are using [Ember Octane](https://blog.emberjs.com/octane-is-here/) now:
* Feel free to migrate existing components to Glimmer Components.
* Write new ones exclusively in it.

Useful Resource : [Ember Octane vs Classic Cheat Sheet](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/)

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/hyper-table-v2/selection.hbs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class={{this.classes}} {{did-insert this.setup}}>
+  <div class="count-container">
+    <i class="fas fa-users"></i>
+    {{#if this.isAllCreatorsSelected}}
```
Comment (Miexil) [inline addon/components/hyper-table-v2/selection.hbs:4]: maybe just `allCreatorsSelected`
If multiple it shouldn't be "is" but rather "are" so the proper English would be "areAllCreatorsSelected" (Nit picking ofc ^^)
IMO, `allCreatorsSelected` is enough

File: addon/components/hyper-table-v2/selection.hbs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      <span>{{{t "hypertable.selection.all_creator_selected" count=@total}}}</span>
```
Comment (Miexil) [inline addon/components/hyper-table-v2/selection.hbs:5]: `hypertable.selection.all_creators_selected` (missing S on creator)

File: addon/components/hyper-table-v2/selection.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    if (this.args.selected > 0) {
+      classes.push('selection-container--visible');
+    }
+
+    return classes.join(' ');
+  }
+
+  @action
+  setup(element: HTMLElement): void {
+    this.componentElement = element;
```
Comment (Miexil) [inline addon/components/hyper-table-v2/selection.ts:30]: Where is `this.componentElement` used ? 🤔

File: tests/integration/components/hyper-table-v2/selection-test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  hooks.beforeEach(function () {
+    this.selected = 1;
+    this.total = 5;
+    this.onClear = sinon.stub();
+    this.onSelectAll = sinon.stub();
+
+    this.intlService = this.owner.lookup('service:intl');
+  });
+
+  test("it doesn't renders when @selected is equals to 0", async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/hyper-table-v2/selection-test.ts:19]: is equal to 0

File: addon/components/hyper-table-v2/selection.ts (hunk 3 above)
Comment (JulienVannier66) [reply addon/components/hyper-table-v2/selection.ts:30]: Old code test, I remove that

**Review by Miexil — APPROVED** [review]

File: addon/components/hyper-table-v2/selection.hbs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<div class={{this.classes}}>
+  <div class="count-container">
+    <i class="fas fa-users"></i>
```
Comment (phndiaye) [inline addon/components/hyper-table-v2/selection.hbs:3]: `far` no ?

File: addon/components/hyper-table-v2/selection.hbs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    {{#if this.allCreatorsSelected}}
+      <span>{{{t "hypertable.selection.all_creators_selected" count=@total}}}</span>
+    {{else}}
+      <span>{{t "hypertable.selection.creator_selected" count=@selected}}</span>
+    {{/if}}
+
```
Comment (phndiaye) [inline addon/components/hyper-table-v2/selection.hbs:9]: extra line

File: addon/components/hyper-table-v2/selection.hbs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  </div>
+  {{#if (not this.allCreatorsSelected)}}
+    <span>-</span>
+    <span class="select-all-container" role="button" {{on "click" this.onSelectAll}}>
+      {{t "hypertable.selection.select_all" count=@total}}
+    </span>
+  {{/if}}
```
Comment (phndiaye) [inline addon/components/hyper-table-v2/selection.hbs:16]: - `unless this.allCreatorsSelected`

**Pull request merged**